admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/data/network_client/NetworkClientImpl.kt

```diff
@@ -0,0 +1,118 @@
+package net.thechance.mena.admin_panel.data.network_client
+
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.auth.providers.BearerTokens
+import io.ktor.client.plugins.auth.providers.bearer
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.defaultRequest
+import io.ktor.client.plugins.logging.LogLevel
+import io.ktor.client.plugins.logging.Logger
+import io.ktor.client.plugins.logging.Logging
+import io.ktor.client.plugins.logging.SIMPLE
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.accept
+import io.ktor.client.request.delete
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.put
+import io.ktor.client.statement.HttpResponse
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.json.Json
+import org.koin.core.annotation.Named
+import org.koin.core.annotation.Provided
+import org.koin.core.annotation.Single
+
+@Single
+class NetworkClientImpl(
+    @Provided @Named("baseUrl") private val baseUrl: String
+) : NetworkClient {
+
+    private val client: HttpClient = buildClient()
+
+    override suspend fun get(
+        urlString: String,
+        requestBuilder: HttpRequestBuilder.() -> Unit
+    ): HttpResponse {
+        return client.get(urlString, requestBuilder)
+    }
```

Comment:
Let's use compact function and named arguments.
```suggestion
    ): HttpResponse = client.get(urlString = urlString, requestBuilder = requestBuilder)
```